src/shogun/preprocessor/RandomFourierGaussPreproc.h

```diff
@@ -85,20 +74,26 @@ class RandomFourierGaussPreproc: public RandomMixin<DensePreprocessor<float64_t>
 		m_dim_output = dims;
 	}
 
+	void set_kernel(const std::shared_ptr<Kernel>& kernel)
+	{
+		require(
+	    std::dynamic_pointer_cast<ShiftInvariantKernel>(kernel),
+	    "Kernel must be shift-invariant");
```

Comment:
"Provided kernel {} is not shift invariant." (or similar, print the name of the given one)